include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -2027,7 +2048,10 @@ __pattern_shift_left(_ExecutionPolicy&& __exec, _Range __rng, oneapi::dpl::__int
     else //2. n < size/2; 'n' parallel copying
     {
         auto __brick = unseq_backend::__brick_shift_left<_ExecutionPolicy, _DiffType>{__size, __n};
-        oneapi::dpl::__par_backend_hetero::__parallel_for(::std::forward<_ExecutionPolicy>(__exec), __brick, __n, __rng)
+        oneapi::dpl::__par_backend_hetero::__parallel_for(
+            oneapi::dpl::__par_backend_hetero::make_wrapped_policy<__shift_left>(
```

Comment:
1) about naming... 
`make_wrapped_policy` - I would name it more shortly - `make_policy` for example. 
A fact that a result policy is wrapped - it is obvious :)
2) M.b. to use a dummy class (for a kernel name) directly - just class name?
`make_policy<class __shift_left>` ?

Reply:
1. It makes sense, but it will lead to a lot of changes in the code now. It can be done in the separate PR. You can open a Github issue to not forget about it.
2. Unfortunately, no. The class should wrap the existing policy to make name unique. So, this class should be templated by policy name. So the kernel name will look like `__shift_left<__exec_name>`

Reply:
1. I agree that it can be done in another patch. So, lets add it into our task list.
2.  Probably, you don't  understand  me... I mean writing code like this:
`oneapi::dpl::execution::make_device_policy<class binarySearch>(oneapi::dpl::execution::dpcpp_default);`

`class binarySearch `is not defined above, one us used just in line.
We have many tests with such writing.

Reply:
We discussed it internally, but I will describe here to make it public. This is a different case. Here we need a class that is a template of templates

Reply:
Discussed offline, I've got the details. We need to keep `template <template <typename>...`  to wrap/keep a passed kernel name (via a policy type).